src/renderers/materialized/materialized.date.control.tsx

```diff
@@ -6,19 +6,20 @@ import { dateControlTester } from '../controls/date.control';
 import { connect, Event } from '../../common/binding';
 
 export class DateControl extends Control<ControlProps, ControlState> {
-
+  private formatter = new Intl.DateTimeFormat(navigator.language, {day: '2-digit', month: '2-digit', year: 'numeric'});
   render() {
     const { classNames, id, visible, enabled, errors, label } = this.props;
+    classNames.input += ' datepicker';
 
     return (
       <div className={classNames.wrapper}>
         <label htmlFor={id} className={classNames.label} data-error={errors}>
           {label}
         </label>
-        <input type='date'
-               value={this.state.value}
-               onChange={(ev: Event<HTMLInputElement>) =>
-                 this.handleChange(new Date(ev.currentTarget.value).toISOString().substr(0, 10))
+        <input type='text'
+               value={this.formatter.format(new Date(this.state.value))}
+               onChange={ev =>
+                 this.handleChange(ev.target.previousSibling.previousSibling.value)
```

Comment:
The date control doesn't seem to work for me (do I need to change any of the files):
* It looks like a regular text field, no date-specific controls
* Can you explain what `ev.target.previousSibling.previousSibling.value` does? Also adding a comment would be good
* `onChange` throws an error in any case
```
Uncaught TypeError: Cannot read property 'value' of null
    at onChange (materialized.date.control.tsx:22)
    at HTMLInputElement.wrappedOnChange (index.js:349)

Reply:
Interesting, that I don't get hit by this issue.
Can you try to increase the timeout in line 50 to `1000` ms?
The `ev.target.previousSibling.previousSibling.value` assumes the dom structure which materialize/piackadate create when providing the date control see [Materialize DatePicker](http://materializecss.com/forms.html#date-picker).

On the long run we need a possibility to get notified when the jsonforms element finishes rendering so that we can run additional code, like the pickadate stuff.

Reply:
as I learned there is one: `componentDidMount()` just overwrite this